pallets/gear/src/lib.rs

```diff
@@ -1733,13 +1756,149 @@ pub mod pallet {
+                session_id,
+                program_id,
+                session_end_block,
+            });
+
+            Ok(().into())
+        }
+
+        /// Appends memory pages to the resume session.
+        ///
+        /// The origin must be Signed and should be the owner of the session.
+        ///
+        /// Parameters:
+        /// - `session_id`: id of the resume session.
+        /// - `memory_pages`: program memory (or its part) before it was paused.
+        #[pallet::call_index(10)]
+        #[pallet::weight(<T as Config>::WeightInfo::resume_session_append(memory_pages.len() as u32))]
+        pub fn resume_session_append(
+            origin: OriginFor<T>,
+            session_id: u64,
+            memory_pages: Vec<(GearPage, PageBuf)>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            ProgramStorageOf::<T>::resume_session_append(
+                session_id,
+                AccountId32::from_origin(who.into_origin()),
+                memory_pages,
+            )?;
+
+            Ok(().into())
+        }
+
+        /// Finishes the program resume session.
+        ///
+        /// The origin must be Signed and should be the owner of the session.
+        ///
+        /// Parameters:
+        /// - `session_id`: id of the resume session.
+        #[pallet::call_index(11)]
+        #[pallet::weight(DbWeightOf::<T>::get().reads(1) + <T as Config>::WeightInfo::resume_session_finish(ProgramStorageOf::<T>::resume_session_page_count(session_id).unwrap_or(0)))]
+        pub fn resume_session_finish(
+            origin: OriginFor<T>,
+            session_id: u64,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            let result = ProgramStorageOf::<T>::resume_session_finish(
+                session_id,
+                AccountId32::from_origin(who.clone().into_origin()),
+                Self::block_number(),
+            )?;
+
+            let task = ScheduledTask::RemoveResumeSession(session_id);
+            let session_end_block = result
+                .start_block
+                .saturating_add(ResumeSessionDurationOf::<T>::get());
+            TaskPoolOf::<T>::delete(session_end_block, task)
+                .unwrap_or_else(|e| unreachable!("Scheduling logic invalidated! {:?}", e));
+
+            let block_author = Authorship::<T>::author()
+                .unwrap_or_else(|| unreachable!("Failed to find block author!"));
+            if let Some((program_id, expiration_block)) = result.info {
+                let task = ScheduledTask::PauseProgram(program_id);
+                TaskPoolOf::<T>::add(expiration_block, task)
+                    .unwrap_or_else(|e| unreachable!("Scheduling logic invalidated! {:?}", e));
+
+                let leftover = CurrencyOf::<T>::repatriate_reserved(
+                    &who,
+                    &block_author,
+                    result.rent_fee.saturated_into(),
```

Comment:
Why the whole rent fee is just burned?

Reply:
this is the rent of the program payed for the new expiration block so the whole value  consumed